File: java/src/automation-root/automation-framework-ui/src/main/java/com/automation/framework/ui/pages/StudentDetailsUi.java
package com.automation.framework.ui.pages;

import com.automation.api.components.Enrollment;
import com.automation.api.pages.StudentDetails;
import com.automation.core.components.FluentUi;
import com.automation.core.logging.Logger;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import java.time.LocalDateTime;

public class StudentDetailsUi extends FluentUi implements StudentDetails {
    public StudentDetailsUi(WebDriver driver) {
        super(driver);
    }

    public StudentDetailsUi(WebDriver driver, Logger logger) {
        super(driver, logger);
    }

    @Override
    public Enrollment[] enrollments() {
        return new Enrollment[0];
    }

    @Override
    public String firstName() {
        return getDriverExtensions().getElement(By.xpath("//dd[2]")).getText().trim();
    }

    @Override
    public String lastName() {
        return null;
    }

    @Override
    public LocalDateTime enrollmentDate() {
        return null;
    }
}

File: csharp/src/AutomationRoot/Automation.Framework.Ui/Pages/StudentDetailsUi.cs
using Automation.Api.Components;
using Automation.Api.Pages;
using Automation.Core.Components;
using Automation.Core.Logging;
using Automation.Extensions.Components;
using OpenQA.Selenium;
using System;

namespace Automation.Framework.Ui.Pages
{
    public class StudentDetailsUi : FluentUi, IStudentDetails
    {
        public StudentDetailsUi(IWebDriver driver)
            : base(driver) { }

        public StudentDetailsUi(IWebDriver driver, ILogger logger)
            : base(driver, logger) { }

        public DateTime EnrollementDate()
        {
            throw new NotImplementedException();
        }

        public IEnrollment[] Enrollments()
        {
            throw new NotImplementedException();
        }

        public string FirstName()
        {
            return Driver.GetElement(By.XPath("//dd[2]")).Text.Trim();
        }

        public string LastName()
        {
            throw new NotImplementedException();
        }
    }
}